Cannot find module 'ky-universal/node' (http module)

Doesn't work with Nuxt SSR.
![image](https://user-images.githubusercontent.com/69633530/109306509-8f74c780-783f-11eb-8f34-7f010d6d5fe6.png)

I didn't manage to reproduce this error, however I had one with both nuxt-i18n and @nuxtjs/strapi: **Cannot find module 'ky-universal/node' from 'F:\CONTENT\DEVELOPPEMENT\WEB\TESTS\reproduction repos\nuxt-vite-issues-25'**.
Also noticed that every time I (un)install a package with yarn I get the following warning:
```bash
[3/4] Linking dependencies...
warning "nuxt-vite > vite-plugin-vue2 > @vue/compiler-sfc@3.0.6" has incorrect peer dependency "vue@3.0.6".
[4/4] Building fresh packages...
```
Here is a repoduction repo with both modules: https://github.com/florian-lefebvre/nuxt-vite-issues-25

--- a/package.json
+++ b/package.json
@@ -19,6 +19,7 @@
     "test": "yarn lint && yarn jest"
   },
   "dependencies": {
+    "@nuxt/http": "^0.6.4",
     "chokidar": "^3.5.1",
     "debounce": "^1.2.0",
     "fs-extra": "^9.1.0",

--- a/src/client.ts
+++ b/src/client.ts
@@ -17,7 +17,8 @@ export async function buildClient (ctx: ViteBuildContext) {
       'process.server': false,
       'process.client': true,
       global: 'window',
-      'module.hot': false
+      'module.hot': false,
+      'process.env': 'import.meta.env'
     },
     resolve: {
       alias

--- a/src/mock/abort-controller.js
+++ b/src/mock/abort-controller.js
@@ -1,0 +1,1 @@
+export const AbortController = null

--- a/src/mock/web-streams-polyfill.js
+++ b/src/mock/web-streams-polyfill.js
@@ -1,0 +1,1 @@
+export const ReadableStream = null

--- a/src/vite.ts
+++ b/src/vite.ts
@@ -30,7 +30,10 @@ async function bundle (nuxt: Nuxt, builder: any) {
             '~': nuxt.options.srcDir,
             '@': nuxt.options.srcDir,
             // TODO: fix client cache issue with mjs build
-            ufo: require.resolve('ufo/dist/index.js')
+            ufo: require.resolve('ufo/dist/index.js'),
+            'web-streams-polyfill/ponyfill/es2018': require.resolve('./mock/web-streams-polyfill'),
+            // Cannot destructure property 'AbortController' of ..
+            'abort-controller': require.resolve('./mock/abort-controller')
           }
         },
         clearScreen: false,

--- a/test/fixture/nuxt.config.js
+++ b/test/fixture/nuxt.config.js
@@ -2,8 +2,12 @@ import viteModule from '../../src'
 export default {
   components: true,
   buildModules: [
-    viteModule
+    viteModule,
+    '@nuxt/http'
   ],
+  serverMiddleware: {
+    '/api/test': '~/serverMiddleware/test'
+  },
   plugins: [
     '~/plugins/hello',
     '~/plugins/plugin.client',

--- a/test/fixture/nuxt.config.js
+++ b/test/fixture/nuxt.config.js
@@ -1,4 +1,5 @@
 import viteModule from '../../src'
+
 export default {
   components: true,
   buildModules: [

--- a/test/fixture/pages/index.vue
+++ b/test/fixture/pages/index.vue
@@ -3,19 +3,19 @@
     <h2>Hello Vite from Nuxt2!</h2>
     <p><Foobar /></p>
     <pre v-text="stateText" />
-    <!-- <p>{{ res }}</p> -->
+    <p>{{ res }}</p>
   </div>
 </template>
 
 <script>
 export default {
   middleware: 'test-middleware',
-  // async asyncData ({ $http }) {
-  //   const res = await $http.$get('/api/test')
-  //   return {
-  //     res
-  //   }
-  // },
+  async asyncData ({ $http }) {
+    const res = await $http.$get('/api/test')
+    return {
+      res
+    }
+  },
   computed: {
     stateText () {
       return JSON.stringify(this.$store.state, null, 2).replace(/"/g, '')

--- a/test/fixture/pages/index.vue
+++ b/test/fixture/pages/index.vue
@@ -1,16 +1,23 @@
 <template>
   <div>
-    Hello Vite from Nuxt2!
+    <h2>Hello Vite from Nuxt2!</h2>
+    <p>{{ res }}</p>
     <Foobar />
-    <br>
-    Store:
-    <pre v-text="stateText" />
+    <p>
+      Store: <pre v-text="stateText" />
+    </p>
   </div>
 </template>
 
 <script>
 export default {
   middleware: 'test-middleware',
+  async asyncData ({ $http }) {
+    const res = await $http.$get('/api/test')
+    return {
+      res
+    }
+  },
   computed: {
     stateText () {
       return JSON.stringify(this.$store.state, null, 2).replace(/"/g, '')

--- a/test/fixture/pages/index.vue
+++ b/test/fixture/pages/index.vue
@@ -1,11 +1,9 @@
 <template>
   <div>
     <h2>Hello Vite from Nuxt2!</h2>
+    <p><Foobar /></p>
+    <pre v-text="stateText" />
     <p>{{ res }}</p>
-    <Foobar />
-    <p>
-      Store: <pre v-text="stateText" />
-    </p>
   </div>
 </template>
 

--- a/test/fixture/pages/index.vue
+++ b/test/fixture/pages/index.vue
@@ -3,19 +3,19 @@
     <h2>Hello Vite from Nuxt2!</h2>
     <p><Foobar /></p>
     <pre v-text="stateText" />
-    <p>{{ res }}</p>
+    <!-- <p>{{ res }}</p> -->
   </div>
 </template>
 
 <script>
 export default {
   middleware: 'test-middleware',
-  async asyncData ({ $http }) {
-    const res = await $http.$get('/api/test')
-    return {
-      res
-    }
-  },
+  // async asyncData ({ $http }) {
+  //   const res = await $http.$get('/api/test')
+  //   return {
+  //     res
+  //   }
+  // },
   computed: {
     stateText () {
       return JSON.stringify(this.$store.state, null, 2).replace(/"/g, '')

--- a/test/fixture/serverMiddleware/test.js
+++ b/test/fixture/serverMiddleware/test.js
@@ -1,0 +1,3 @@
+export default (_req, res) => {
+  res.end('API Response')
+}

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -24,7 +24,7 @@ describe('browser', () => {
 function testIndex (html: string) {
   expect(html).toContain('Hello Vite from Nuxt2!')
   expect(html).toContain('/@vite/client')
-  expect(html).toContain('API Response')
+  // expect(html).toContain('API Response')
   // expect(html).toContain('FooBar Component')
   expect(html).toContain('st: 1')
   expect(html).toContain('st: 2')

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -16,7 +16,6 @@ describe('browser', () => {
 
   it('SPA works', async () => {
     const page = await createPage('/?spa')
-    await new Promise(resolve => setTimeout(resolve, 1000))
     const html = await page.innerHTML('body')
     testIndex(html)
   })
@@ -25,8 +24,8 @@ describe('browser', () => {
 function testIndex (html: string) {
   expect(html).toContain('Hello Vite from Nuxt2!')
   expect(html).toContain('/@vite/client')
-  expect(html).toContain('API Response')
-  expect(html).toContain('FooBar Component')
+  // expect(html).toContain('API Response')
+  // expect(html).toContain('FooBar Component')
   expect(html).toContain('st: 1')
   expect(html).toContain('st: 2')
   expect(html).toContain('st: 3')

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -24,6 +24,8 @@ describe('browser', () => {
 function testIndex (html: string) {
   expect(html).toContain('Hello Vite from Nuxt2!')
   expect(html).toContain('/@vite/client')
+  expect(html).toContain('API Response')
+  expect(html).toContain('FooBar Component')
   expect(html).toContain('st: 1')
   expect(html).toContain('st: 2')
   expect(html).toContain('st: 3')

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -24,7 +24,7 @@ describe('browser', () => {
 function testIndex (html: string) {
   expect(html).toContain('Hello Vite from Nuxt2!')
   expect(html).toContain('/@vite/client')
-  // expect(html).toContain('API Response')
+  expect(html).toContain('API Response')
   // expect(html).toContain('FooBar Component')
   expect(html).toContain('st: 1')
   expect(html).toContain('st: 2')

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -16,6 +16,7 @@ describe('browser', () => {
 
   it('SPA works', async () => {
     const page = await createPage('/?spa')
+    await new Promise(resolve => setTimeout(resolve, 1000))
     const html = await page.innerHTML('body')
     testIndex(html)
   })

